### PR TITLE
Google Drive Integration

### PR DESCRIPTION
**Pull request by efuquen**
Adds an open and save option for Google Drive on the navigation bar, which opens a custom Google Drive file browser to navigate to a file to open or a folder to save the current file. When clicking the save icon on an already open file it will automatically save to the same Google Drive if that was the source of the current file.

### CODE REVIEW

File: src/file_handler/file_saver.js [hunk 1]
```diff
@@ -84,11 +84,15 @@ cwc.fileHandler.FileSaver.prototype.saveFileAs = function() {
 /**
  * @export
```
Comment (MarkusBordihn) [inline src/file_handler/file_saver.js]: Please add missing @param and rename param "saveFile" to "save_file".

File: src/ui/account.js [hunk 2]
```diff
@@ -190,39 +190,29 @@ cwc.ui.Account.prototype.setAuthentication = function(authenticated) {
...
 /**
- * @param {Object} args The args object supports the following properties:
- *   - callback
- *   - content
- *   - header
- *   - method
- *   - param
- *   - path
- *   - token
+ * @param {?} opts
```
Comment (MarkusBordihn) [inline src/ui/account.js]: Please add an short list over the supported "opts".

File: src/ui/gdrive/gdrive.js [hunk 3]
```diff
@@ -27,9 +27,31 @@ goog.require('goog.dom.dataset');
...
+goog.require('goog.ui.Dialog.EventType');
 goog.require('goog.ui.Menu');
 goog.require('goog.ui.MenuItem');
 
+var PAGE_SIZE = 1000;
+var ORDER_BY = 'folder,modifiedTime desc,name';
+var FILE_FIELDS = (
+    'files(id,mimeType,parents,iconLink,modifiedTime,name,owners)');
+var FOLDER_MIME_TYPE = 'application/vnd.google-apps.folder';
+var EXT_TO_MIME_TYPE = {
```
Comment (MarkusBordihn) [inline src/ui/gdrive/gdrive.js]: Please move these "global" definitions to an new file under "src/config/gdrive.js" with the namespace 'cwc.config.GDrive...'.

File: src/ui/gdrive/gdrive.js [hunk 4]
```diff
@@ -92,46 +135,94 @@ cwc.ui.GDrive.prototype.decorate = function(node, opt_prefix) {
...
     'q': 'mimeType = \'' + this.mimeType + '\''
   }, fileEvent);
 };
 
 
+cwc.ui.GDrive.prototype.openDialog = function() {
+  this.dialogType = 'open';
+  this.getMyFiles();
+};
+
```
Comment (MarkusBordihn) [inline src/ui/gdrive/gdrive.js]: Please add simple JsDoc.

File: src/ui/gdrive/gdrive.js [hunk 5]
```diff
@@ -92,46 +135,94 @@ cwc.ui.GDrive.prototype.decorate = function(node, opt_prefix) {
...
   var fileEvent = this.handleFileList.bind(this);
   this.getFiles({
-    'maxResults': '100',
+    'pageSize': PAGE_SIZE,
     'q': 'mimeType = \'' + this.mimeType + '\''
   }, fileEvent);
 };
 
 
+cwc.ui.GDrive.prototype.openDialog = function() {
```
Comment (MarkusBordihn) [inline src/ui/gdrive/gdrive.js]: Please add simple JsDoc.

File: src/ui/gdrive/gdrive.js [hunk 6]
```diff
@@ -92,46 +135,94 @@ cwc.ui.GDrive.prototype.decorate = function(node, opt_prefix) {
...
+    'pageSize': PAGE_SIZE,
+    'orderBy': ORDER_BY,
+    'fields': FILE_FIELDS,
+    'q': (ACCEPTED_MIME_TYPE_QUERY + ' and \'me\' in owners and \'root\' ' +
+       'in parents and trashed = false')
   }, fileEvent);
 };
 
 
+cwc.ui.GDrive.prototype.getSubFolder = function(file, trashed) {
```
Comment (MarkusBordihn) [inline src/ui/gdrive/gdrive.js]: Please add simple JsDoc.

File: src/ui/gdrive/gdrive.js [hunk 7]
```diff
@@ -178,27 +286,69 @@ cwc.ui.GDrive.prototype.getTrashFiles = function() {
  * Updates the GDrive filelist with the new files.
  * @param {Object} files Filelist with the result of the search.
  */
-cwc.ui.GDrive.prototype.updateFileList = function(files) {
+cwc.ui.GDrive.prototype.updateFileList = function(files, dialog) {
```
Comment (MarkusBordihn) [inline src/ui/gdrive/gdrive.js]: Please add @param for dialog.

File: src/ui/helper.js [hunk 8]
```diff
@@ -109,6 +109,20 @@ cwc.ui.Helper.getMenuItem = function(name, opt_title, opt_func) {
 };
 
 
+cwc.ui.Helper.getListItem = function(name, opt_func) {
```
Comment (MarkusBordihn) [inline src/ui/helper.js]: Please add simple JsDoc.

**Comment by efuquen** [comment]
PTAL

**Pull request merged**